cpg_pipes/types.py

```diff
@@ -204,3 +174,51 @@ def parse(str_val: str) -> 'SequencingType':
                 f'Available: {list(str_to_val.keys())}'
             )
         return str_to_val[str_val]
+
+
+class AlignmentInput:
+    """
+    Alignment input can be an indexed CRAM file, or a list of FASTQ file pairs.
+    """
+    def __init__(
+        self, 
+        data: Union[CramPath, FastqPairs],
+        sequencing_type: SequencingType,
+    ):
+        self.data = data
+        self.is_cram_or_bam = isinstance(data, CramPath)
+        self.sequencing_type = sequencing_type
+        
+    def __str__(self):
+        return f'{self.sequencing_type.value}:{str(self.data)}'
+
+    def exists(self) -> bool:
+        """
+        Check if all files in alignment inputs exist.
+        """
+        if isinstance(self.data, CramPath):
+            return utils.exists(self.data.path)
+        else:
+            return all(
+                utils.exists(pair.r1) and utils.exists(pair.r2) 
+                for pair in self.data
+            )
+
+    def compact_file_paths(self) -> str:
+        """
+        Compact representation of file paths. 
+        For CRAM, it's simply path to CRAM. 
+        For FASTQ pairs, it's glob string to find all FASTQ files.
+        """
```

Comment:
Not a super big deal, but this would be cool as a static method, with doctests

Reply:
Ah, so doctests only work with static method? I can't get it working with a class method, it throws `Module [...] has no class "AlignmentInput"` :(